sim/hunter/kill_shot.go

```diff
@@ -45,7 +47,11 @@ func (hunter *Hunter) registerKillShotSpell() {
 		BonusCoefficient: 1,
 
 		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
-			if sim.IsExecutePhase20() {
+			if hasStrikersPursuit2P &&  sim.CurrentTime > sim.Encounter.Duration / 2 {
+				spell.CD.Set(sim.CurrentTime + spell.CD.TimeToReady(sim)/2)
+			}
```

Comment:
Could this be done with an `OnCastComplete` callback in the set bonus code?

Reply:
Yeah that's a much better solution... updated!